tests/store/tracking/test_file_store.py

```diff
@@ -724,21 +755,60 @@ def test_get_deleted_runs(self):
         assert len(deleted_runs) == 1
         assert deleted_runs[0] == run_id
 
-    def test_create_run_appends_to_artifact_uri_path_correctly(self):
+    @staticmethod
+    def _assert_create_run_appends_to_artifact_uri_path_correctly(cases):
+        for artifact_root_uri, expected_artifact_uri_format in cases:
+            with TempDir() as tmp:
+                fs = FileStore(tmp.path(), artifact_root_uri)
+                exp_id = fs.create_experiment("exp")
+                run = fs.create_run(
+                    experiment_id=exp_id, user_id="user", start_time=0, tags=[], run_name="name"
+                )
+                cwd = Path.cwd().as_posix()
+                drive = Path.cwd().drive
+                if is_local_os_windows() and expected_artifact_uri_format.startswith("file:"):
+                    cwd = f"/{cwd}"
+                    drive = f"{drive}/"
+                assert run.info.artifact_uri == expected_artifact_uri_format.format(
+                    e=exp_id, r=run.info.run_id, cwd=cwd, drive=drive
+                )
+
+    @pytest.mark.skipif(not is_local_os_windows(), reason="This test only passes on Windows")
+    def test_create_run_appends_to_artifact_local_path_correctly_on_windows(self):
         cases = [
-            ("path/to/local/folder", "path/to/local/folder/{e}/{r}/artifacts"),
+            ("path/to/local/folder", "file://{cwd}/path/to/local/folder/{e}/{r}/artifacts"),
+            ("/path/to/local/folder", "file:///{drive}path/to/local/folder/{e}/{r}/artifacts"),
+            ("#path/to/local/folder?", "file://{cwd}/{e}/{r}/artifacts#path/to/local/folder?"),
+        ]
+        self._assert_create_run_appends_to_artifact_uri_path_correctly(cases)
+
+    @pytest.mark.skipif(is_local_os_windows(), reason="This test fails on Windows")
+    def test_create_run_appends_to_artifact_local_path_correctly(self):
+        cases = [
+            ("path/to/local/folder", "{cwd}/path/to/local/folder/{e}/{r}/artifacts"),
             ("/path/to/local/folder", "/path/to/local/folder/{e}/{r}/artifacts"),
-            ("#path/to/local/folder?", "#path/to/local/folder?/{e}/{r}/artifacts"),
-            ("file:path/to/local/folder", "file:path/to/local/folder/{e}/{r}/artifacts"),
-            ("file:///path/to/local/folder", "file:///path/to/local/folder/{e}/{r}/artifacts"),
+            ("#path/to/local/folder?", "{cwd}/#path/to/local/folder?/{e}/{r}/artifacts"),
+        ]
+        self._assert_create_run_appends_to_artifact_uri_path_correctly(cases)
+
+    def test_create_run_appends_to_artifact_uri_path_correctly(self):
+        cases = [
+            ("file:path/to/local/folder", "file://{cwd}/path/to/local/folder/{e}/{r}/artifacts"),
+            (
+                "file:///path/to/local/folder",
+                "file:///{drive}path/to/local/folder/{e}/{r}/artifacts",
```

Comment:
I think this looks confusing because Unix systems don't have a drive.

Reply:
done and also, converted all artifacts tests from cases dict to parameterized pytests. It was just painful to find which tests failed one by one since it fails on first assert from that dict entry.